apps/www/src/registry/default/plate-ui/table-element.tsx

```diff
@@ -114,7 +118,66 @@ const TableFloatingToolbar = React.forwardRef<
   const readOnly = useReadOnly();
   const selected = useSelected();
   const editor = useEditorState();
-  const open = !readOnly && selected && isCollapsed(editor.selection);
+  // const open = !readOnly && selected && isCollapsed(editor.selection);
+
+  const collapsed = !readOnly && selected && isCollapsed(editor.selection);
+  const expanded = !readOnly && selected && isExpanded(editor.selection);
+  const open = !readOnly && selected;
+
+  const cellEntries = getTableGridAbove(editor, { format: 'cell' });
+
+  const canUnmerge =
+    collapsed &&
+    cellEntries &&
+    cellEntries.length === 1 &&
+    ((cellEntries[0][0] as any)?.colSpan > 1 ||
+      (cellEntries[0][0] as any)?.rowSpan > 1);
```

Comment:
This should be moved to `useTableMergeState` (a new function in table package), please reuse the existing pattern

Reply:
Done